CHANGELOG.md

```diff
@@ -1,122 +1,124 @@
+- You can invoke the `GenerateSchema`, `GenerateSchemaAndSnapshots`, and `CookAndGenerateSchema` commandlets with the `-AdditionalSchemaCompilerArguments="..."` command line switch to output additional compiled schema formats. If you don't provide this switch, the output contains only the schema descriptor. This switch's value should be a subset of the arguments that you can pass to the schema compiler directly (for example `--bundle_out="path/to/bundle.sb"`). You can see a full list of possible values in the [schema compiler documentation](https://docs.improbable.io/reference/14.2/shared/schema/introduction#schema-compiler-cli-reference)
+- We've added the `AllowUnresolvedParameters` function flag that disables warnings for processing RPCs with unresolved parameters. To enable this flag, use Blueprints, or add a tag to the `UFUNCTION` macro.
+- There is now a warning if you launch a cloud deployment with the `manual_worker_connection_only` flag set to true.
+- We now support server travel for single server game worlds. Does not currently support zoning or off-loading.
+- We've improved the workflow relating to schema generation issues and launching local builds. There is now a warning if you try to run a local deployment after a schema error.
+- `DeploymentLauncher` can now launch a simulated player deployment independently from the target deployment.
 Usage: `DeploymentLauncher createsim <project-name> <assembly-name> <target-deployment-name> <sim-deployment-name> <sim-deployment-json> <sim-deployment-region> <num-sim-players> <auto-connect>`
 - Added `HeartbeatTimeoutWithEditorSeconds` and use it if WITH_EDITOR is defined to prevent workers disconnecting when debugging while running in editor.
-- Added the `bAsyncLoadNewClassesOnEntityCheckout` setting to SpatialGDKSettings that allows loading new classes asynchronously when checking out entities. This is off by default.
-- Added `IndexYFromSchema` functions for the `Coordinates`, `WorkerRequirementSet`, `FRotator`, and `FVector` classes. Remapped the `GetYFromSchema` functions for the same classes to invoke `IndexYFromSchema` internally, in line with other implementations of the pattern.
-- The logic responsible for taking an Actor and generating the array of Components that represents it as an Entity in SpatialOS has been extracted into `EntityFactory`.
-- Clients will now validate schema against the server and log a warning if they do not match.
-- Entries in the SchemaDatabase are now sorted to improve efficiancy when browsing the asset in the editor. (DW-Sebastien)
-- Load Balancing Strategies and Locking Strategies can be set per-level using SpatialWorldSettings.
+- We've added the `bAsyncLoadNewClassesOnEntityCheckout` setting to SpatialGDKSettings that allows loading new classes asynchronously when checking out entities. This is off by default.
```

Comment:
What’s “SpatialGDKSettings”?

Reply:
What does “checking out” mean here? We got rid of this term a while ago in the SpatialOS docs (or does it have a different meaning here?)

Reply:
> `SpatialGDKSettings` is the settings class where we store all of our settings. They are exposed to the user via the class or via the `SpatialOS Settings` window in the editor.

In that case, would it make more sense to talk about the SpatialOS Settings window? E.g. "You can now enable X from the SpatialOS Settings window" (where X is however `bAsyncLoadNewClassesOnEntityCheckout` appears in the window)? Or is there a reason why we can't/shouldn't do this?

> Checking out means checking out, when the runtime sends you an entity you are "checking it out", how can we get rid of this term? We use it everywhere.

Laura and I had a long discussion about this term with Sean last year, when we overhauled the docs on authority and interest (in the current Overview section). We realised that it wasn't clear in our docs what "check out" meant, in relation to interest, and that it could be a confusing term as it has lots of other meanings (e.g. checking out a branch on GitHub, checking something out as in "looking at something").

After much discussion, we realised that we didn't need to talk about "checking out" components, and could instead talk about a worker instance "receiving updates about" components (or, in non-GDK SpatialOS terms, having _active read access_ to components). "Receive updates about" is clear - it spells out what's actually happening - whereas "checks out" doesn't really tell you what's going on.

Similarly, we replaced "checkout radius" with "interest radius" or "area of interest". It's not the radius within which a worker instance checks out (receives updates about) components, because _whether_ it receives updates about components in this radius depends on other factors (read access permission/component filters).

Sean did mention that there was a nuance about "checking out" - possibly something to do with the initial update received about a component - but I think we decided that we didn't need to make this distinction.

(I know it's in the code, so we're clearly not going to be able to get rid of it completely.)

Reply:
1- I'm not so sure about that. I prefer mentioning the class itself rather than the name of the editor tab.
2- Seems like I've lost this one. I agree that checking out refers to the first time you see an entity/component and not just the process of being able to read that entity/component data. So technically they are different.